src/app/functions/server/featureFlags.ts

```diff
@@ -7,10 +7,22 @@ import {
   FeatureFlag,
   getFeatureFlagByName,
 } from "../../../db/tables/featureFlags";
+import { getServerSession } from "next-auth";
+import { authOptions } from "../../api/utils/auth";
+
+export type FeatureFlagsEnabled = {
+  FreeBrokerScan: boolean;
+  PremiumBrokerRemoval: boolean;
+};
+
+export async function isFlagEnabled(name: string): Promise<boolean> {
+  const session = await getServerSession(authOptions);
+  if (!session?.user?.email) {
+    throw new Error("No session");
+  }
 
-export async function getFlag(name: string): Promise<FeatureFlag | undefined> {
   if (!name) {
-    const err = new Error("No name provided to getFlag");
+    const err = new Error("No name provided to isFlagEnabled");
     captureException(err);
     throw err;
   }
```

Comment:
Do we expect this function to be called by non-TypeScript code? Otherwise this check is redundant with the type.